packages/salesforcedx-vscode-core/src/util/configUtil.ts

```diff
@@ -6,79 +6,110 @@
-    if (isUndefined(source) || source === ConfigSource.Local) {
-      try {
-        const rootPath = getRootWorkspacePath();
-        const myLocalConfig = await ConfigFile.create({
-          isGlobal: false,
-          rootFolder: path.join(rootPath, '.sfdx'),
-          filename: 'sfdx-config.json'
-        });
-        const localValue = myLocalConfig.get(key);
-        if (!isNullOrUndefined(localValue)) {
-          return localValue;
-        }
-      } catch (err) {
-        telemetryService.sendException('get_config_value_local', err.message);
-        return undefined;
-      }
-    }
-    if (isUndefined(source) || source === ConfigSource.Global) {
-      try {
-        const aggregator = await ConfigAggregator.create();
-        const globalValue = aggregator.getPropertyValue(key);
-        if (!isNullOrUndefined(globalValue)) {
-          return globalValue;
-        }
-      } catch (err) {
-        telemetryService.sendException('get_config_value_global', err.message);
-        return undefined;
-      }
-    }
-    return undefined;
+  /*
+   * The User-configured API version is set by the user, and is used to
+   * override the API version that is otherwise gotten from the authenticated
+   * Org in some cases, such as when deploying metadata.
+   */
+  public static async getUserConfiguredApiVersion(): Promise<
+    string | undefined
+  > {
+    const configAggregator = await ConfigAggregatorProvider.getInstance().getConfigAggregator();
+    const apiVersion = configAggregator.getPropertyValue(
+      OrgConfigProperties.ORG_API_VERSION
+    );
+    return apiVersion ? String(apiVersion) : undefined;
   }
 
-  public static async getUserConfiguredApiVersion() {
-    const apiVersion = await ConfigUtil.getConfigValue('apiVersion');
-    return apiVersion ? String(apiVersion) : undefined;
+  public static async getDefaultUsernameOrAlias(): Promise<string | undefined> {
+    const configAggregator = await ConfigAggregatorProvider.getInstance().getConfigAggregator();
+    const defaultUsernameOrAlias = configAggregator.getPropertyValue(
+      OrgConfigProperties.TARGET_ORG
+    );
+    return defaultUsernameOrAlias ? String(defaultUsernameOrAlias) : undefined;
+  }
+
+  public static async isGlobalDefaultUsername(): Promise<boolean> {
+    const configSource: ConfigSource = await ConfigUtil.getConfigSource(
+      OrgConfigProperties.TARGET_ORG
+    );
+    return configSource === ConfigSource.Global;
+  }
+
+  /*
+   * Currently, the docs tell Users to manually create this entry in the .sfdx
+   * configuration file.  For that reason, getTemplatesDirectory uses the
+   * SfdxConfigAggregator specifically.
+   */
+  public static async getTemplatesDirectory(): Promise<string | undefined> {
+    const sfdxConfigAggregator = await ConfigAggregatorProvider.getInstance().getSfdxConfigAggregator();
+    const templatesDirectory = sfdxConfigAggregator.getPropertyValue(
+      SfdxPropertyKeys.CUSTOM_ORG_METADATA_TEMPLATES
```

Comment:
Debugging through this the currect setup results in always getting undefined back.  If I update it to use `OrgConfigProperties.ORG_CUSTOM_METADATA_TEMPLATES` we are in business.

Reply:
Regular Config Agg with the new key works.  Removed the sfdx config agg part of the provider!  Yay for simpler